core/src/main/scala/pureconfig/package.scala

```diff
@@ -143,9 +166,13 @@ package object pureconfig {
    */
   @throws[ConfigReaderException[_]]
   def loadConfigOrThrow[Config](path: Path)(implicit conv: ConfigConvert[Config], ct: ClassTag[Config]): Config = {
-    ConfigFactory.invalidateCaches()
-    val config = ConfigFactory.load(ConfigFactory.parseFile(path.toFile))
-    getResultOrThrow[Config](loadConfig[Config](config)(conv))
+    val errorOrConfig =
+      for {
+        _ <- invalidateCaches().right
+        rawConfig <- loadFile(path).right
+        config <- loadConfig[Config](rawConfig)(conv).right
+      } yield config
+    getResultOrThrow[Config](errorOrConfig)
```

Comment:
Having had a better look at this, this is definitely a mistake. We should call `ConfigValueLocation(rawConfig.root())` [here](https://github.com/melrief/pureconfig/pull/161/files#diff-12918ae14aad73e551e678d26010a1deR46) and [here](https://github.com/melrief/pureconfig/pull/161/files#diff-12918ae14aad73e551e678d26010a1deR79) (and reuse this code on the `loadConfigOrThrow` methods).

I don't know if you prefer doing this in the context of this PR or in a separate one.